packages/playground/resolve/__tests__/resolve.spec.ts

```diff
@@ -17,7 +17,9 @@ test('deep import with exports field', async () => {
 })
 
 test('deep import with query with exports field', async () => {
-  expect(await page.textContent('.exports-deep-query')).not.toMatch('fail')
+  expect(await page.textContent('.exports-deep-query')).toMatch(
+    isBuild ? /base64/ : '/exports-path/deep.json'
+  )
```

Comment:
It's unclear to me how only this test change covers the `*.html?url` and non-ESM `*.js?url` cases which failed in [two](https://github.com/vitejs/vite/pull/7097#discussion_r815299116) [different](https://github.com/vitejs/vite/pull/7097#discussion_r815295452) ways before this PR.

It seems like we should add test cases for each of those scenarios, example tests in https://github.com/vitejs/vite/pull/7097

Reply:
Thoses cases failed because `import 'something?url'` was not returning url. They were treated as if `?url` does not exist.
So I changed this test case to check if the exported value is url.

Reply:
Your explanation makes sense. I'd be favor of a comment to explain what we're expecting here as it was a little obtuse for me to follow ⏩